lib/internal/url.js

```diff
@@ -864,6 +866,11 @@ function update(url, params) {
     ctx.search = '?' + serializedParams;
   } else {
     ctx.search = '';
+
+    // Potentially strip trailing spaces from an opaque path
+    if (ctx.hasOpaquePath && !ctx.hasHash) {
+      ctx.pathname = ctx.pathname.trimEnd();
```

Comment:
`trimEnd` would trim off more characters than just U+0020 SPACE characters, so we should not use it here.

Reply:
Nice catch!